### ISSUE TITLE
Samples for version 0.9.x

### ISSUE THREAD

**Issue by majioa** [google/google-api-ruby-client#538]
Hello,

it seems that many of the code has been rewritten for version 0.9.x since 0.8.x so the samples from here  https://github.com/google/google-api-ruby-client-samples/tree/master/service_account aren't supported. When I saw the embedded into this repo `google-api-ruby-client
` I saw the similar samples, but many config parameters passed via ENV vars like:

    GOOGLE_CLIENT_SECRETS
    GOOGLE_CREDENTIAL_STORE
    GOOGLE_CLIENT_ID
    GOOGLE_CLIENT_SECRET
    GOOGLE_API_KEY

with no format description of the vars provided.

So the question is how to use these vars to make user authentication/authorization with a public key for the OAuth2 in 0.9.x?

UPD:

Also the plain run the app:

    google-api-samples

raises the exception:

    samples/cli/google-api-samples:24:in `<class:App>': uninitialized constant App::Samples (NameError)
	from samples/cli/google-api-samples:17:in `<main>'

What is it?

**Comment by neycodes** [comment]
Also looking to solve the authentication issue. Any help would be appreciated since the docs are now obsolete.

**Comment by bootstraponline** [comment]
Documentation on how to authorize correctly would be helpful. The samples just error with `~/.config/google/client_secrets.json` not found.

**Comment by vwoo** [comment]
I got this running a month ago with google-api-client ~> 0.11.1. Though would like official docs in case I missed important details.

Here is what I did for service account authentication. This works well if you don't need to do things on behalf of other user accounts.

- goto https://console.developers.google.com/
- Under 'Library' in the left sidebar, search for the API you want to use then click on the top ENABLE link
-  Under 'Credentials' in the left sidebar, click on "Create credentials" then "Service account key"
- Download the json file
- In your ruby project, specify the filepath to said json file as an env var named GOOGLE_APPLICATION_CREDENTIALS so it is the equivalent of:

`ENV['GOOGLE_APPLICATION_CREDENTIALS'] = '/path/to/downloaded_file.json'`

How you do this will depend on your own setup.

- Haven't tried it but you might not need to specify the filepath if you place the json file in ~/.config/google/client_secrets.json
